### ISSUE TITLE
Use consistent prerequisites in how-to guides

### ISSUE THREAD

**Issue by ptgott** [gravitational/teleport#21541]
## Details

Let's ensure that our how-to guides have consistent prerequisites, both in terms of:

- Having a consistent level of detail/completeness
- Starting in a consistent place

In particular, we want to make sure that we follow this item in our [style guide](https://goteleport.com/docs/contributing/documentation/style-guide/#expectations):

> Do not recreate content where unnecessary. Actions that are prerequisites to several topics but not directly related to those topic should be documented in a single location and referenced in a Prerequisites section at the top of the page.

## Applies To

All of our how-to guides (`grep -Rl "## Step 1" docs/pages`).

It's also worth noting that, if our [`edition-prereqs-tabs.mdx`](https://github.com/gravitational/teleport/blob/23481cd831c0faecc0632ff648e3c7a7f71da94e/docs/pages/includes/edition-prereqs-tabs.mdx) and [`commercial-prereqs-tabs.mdx`](https://github.com/gravitational/teleport/blob/master/docs/pages/includes/commercial-prereqs-tabs.mdx) partials are the closest thing we have to a consistent approach, a number of our how-to guides do not include one of these partials:

```
$ echo "\n" && find docs/pages -name "*.mdx" -not -wholename "*/includes/*" -exec awk '
BEGIN{prereqs=0;howto=0;}
/^## Step 1/{howto=1}
/(edition-prereqs-tabs\.mdx|commercial-prereqs-tabs\.mdx)/{prereqs=1}
END{if (howto==1 && prereqs==0) print FILENAME}' {} \;


docs/pages/database-access/guides/redshift-serverless.mdx
docs/pages/machine-id/guides/kubernetes.mdx
docs/pages/machine-id/guides/jenkins.mdx
docs/pages/machine-id/guides/applications.mdx
docs/pages/machine-id/guides/databases.mdx
docs/pages/machine-id/guides/ansible.mdx
docs/pages/access-controls/guides/passwordless.mdx
docs/pages/deploy-a-cluster/helm-deployments/gcp.mdx
docs/pages/deploy-a-cluster/helm-deployments/aws.mdx
docs/pages/deploy-a-cluster/helm-deployments/kubernetes-cluster.mdx
docs/pages/deploy-a-cluster/helm-deployments/custom.mdx
docs/pages/deploy-a-cluster/helm-deployments/digitalocean.mdx
docs/pages/desktop-access/active-directory.mdx
docs/pages/desktop-access/active-directory-manual.mdx
docs/pages/contributing/documentation/how-to-contribute.mdx
docs/pages/management/admin/trustedclusters.mdx
docs/pages/management/admin/daemon.mdx
docs/pages/management/admin/upgrading-the-teleport-binary.mdx
docs/pages/management/operations/tls-routing.mdx
docs/pages/management/operations/proxy-peering.mdx
docs/pages/management/guides/ssh-key-extensions.mdx
docs/pages/management/guides/docker.mdx
docs/pages/management/guides/joining-nodes-aws-ec2.mdx
docs/pages/try-out-teleport/linux-server.mdx
docs/pages/try-out-teleport/docker-compose.mdx
docs/pages/try-out-teleport/local-kubernetes.mdx
docs/pages/try-out-teleport/digitalocean.mdx
docs/pages/server-access/guides/recording-proxy-mode.mdx
docs/pages/server-access/guides/ansible.mdx
docs/pages/api/getting-started.mdx
docs/pages/choose-an-edition/teleport-cloud/getting-started.mdx
docs/pages/choose-an-edition/teleport-enterprise/gcp-kms.mdx
docs/pages/choose-an-edition/teleport-enterprise/getting-started.mdx
docs/pages/choose-an-edition/teleport-enterprise/hsm.mdx
docs/pages/application-access/jwt/elasticsearch.mdx
docs/pages/application-access/cloud-apis/aws-console.mdx
```
## Related Issues
- https://github.com/gravitational/teleport/issues/18761

**Comment by ptgott** [comment]
We should also re-examine the `edition-prereqs-tabs.mdx` and `commercial-prereqs-tabs.mdx` partials. For example, we might want to use parameters within these partials to include a version warning, e.g.,:

```text
{{ minversion=(!teleport.version!) }}

You need a Teleport cluster v{{ minversion }} or above.
```

**Comment by alexfornuto** [comment]
> hsm

I'd love to include minimum version warnings in a parameter call for this partial. I think what we'd need to make it work is to be able to not display the entire sentence unless the parameter was provided.

**Comment by alexfornuto** [comment]
@ptgott I think we can call this one closed by #23394, unless you want to continue tracking work to modify the prereqs-tabs partials with minimum version warnings. As I said above we'd probably want to make it an all or nothing inclusion, unless we just set the default to the minimum supported version (not technically true, but also kind of true).
